### PR TITLE
Config

### PR DESCRIPTION
**Pull request by mayabrandi**
Removing environment vareable dependency and instead adding option for config. The  following arguments are added to the base command:
```
@click.option("-c", "--config", type=click.File(), help="Path to config file")
@click.option("-u", "--db-uri", type=str, default='mongodb://localhost:27030', help="If no config")
@click.option("-n", "--db-name", type=str, default='vogue-stage', help="If no config")
@click.option("-d", "--debug", type=int, help="0/1", default=1, help="If no config")
@click.option("-s", "--secret-key", type=str, default='hej', help="If no config")
```

**How to prepare for test**:
- [x] ssh to clinical-db
- [x] install on stage of clinical-db:
`bash servers/resources/clinical-db.sclifelab.se/update-vogue-stage.sh [THIS-BRANCH-NAME]`

**How to test**:
- [x] run: ```vogue load sample -d 1```

create a test config test.yaml with the content:
```
DB_URI: 'mongodb://localhost:27030'
DB_NAME: 'vogue-stage'
DEBUG: 0
SECRET_KEY: 'hej'
```
- [x]  run: ```vogue -c ../config/vogue.yaml load sample -d 1```

**Expected test outcome**:
- [x] check that both commands run without errors. They should load data for samples modified the latest day.
- [x] Take a screenshot and attach or copy/paste the output.
- [x] check that all travis tets pass

**Review:**
- [x] code approved by @hassanfa 
- [x] tests executed by @mayabrandi 
- [x] "Merge and deploy" approved by @mayabrandi 
Thanks for filling in who performed the code review and the test!

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by mayabrandi** [comment]
Thanks @hassanfa 
Would you like to review the code as well?

**Comment by hassanfa** [comment]
> Thanks @hassanfa
> Would you like to review the code as well?

right 😁 . I forgot about that part. Looking into it right now.

File: vogue/commands/base.py [hunk 1]
```diff
@@ -23,11 +25,27 @@
 @click.version_option(__version__)
 @click.group(cls=FlaskGroup,
              create_app=create_app,
-             add_default_commands=True,
+             add_default_commands=True, 
              invoke_without_command=False,
              add_version_option=False)
-def cli(**_):
+@click.option("-c", "--config", type=click.File(), help="Path to config file")
+@click.option("-u", "--db-uri", type=str, default='mongodb://localhost:27030', help="If no config")
```
Comment (hassanfa) [inline vogue/commands/base.py:32]: Click by default is String. But no harm done keep it as it is

Comment (hassanfa) [inline vogue/commands/base.py:32]: I suggest making help a bit more descriptive:

```suggestion
@click.option("-u", "--db-uri", type=str, default='mongodb://localhost:27030', help="Set db uri if no config is provided")
```

File: vogue/commands/base.py [hunk 2]
```diff
@@ -23,11 +25,27 @@
...
+@click.option("-n", "--db-name", type=str, default='vogue-stage', help="If no config")
```
Comment (hassanfa) [inline vogue/commands/base.py:33]:
```suggestion
@click.option("-n", "--db-name", type=str, default='vogue-stage', help="Set db name to connect if no config is provided.")
```

File: vogue/commands/base.py [hunk 3]
```diff
@@ -23,11 +25,27 @@
...
+@click.option("-d", "--debug", type=click.Choice(["0", "1"]), default="1", help="If no config")
```
Comment (hassanfa) [inline vogue/commands/base.py:34]:
```suggestion
@click.option("-d", "--flask-debug", type=click.Choice(["0", "1"]), default="1", help="Debug mode for Flask")
```

File: vogue/commands/base.py [hunk 4]
```diff
@@ -23,11 +25,27 @@
...
+@click.option("-s", "--secret-key", type=str, default='hej', help="If no config")
```
Comment (hassanfa) [inline vogue/commands/base.py:35]: I don't know a good suggestion for the help text here. But something more descriptive.

File: vogue/commands/base.py [hunk 5]
```diff
@@ -23,11 +25,27 @@
...
+@with_appcontext
+def cli(config, db_uri, db_name, debug, secret_key):
```
Comment (hassanfa) [inline vogue/commands/base.py:37]:
```suggestion
def cli(config, db_uri, db_name, flask_debug, secret_key):
```

File: vogue/commands/base.py [hunk 6]
```diff
@@ -23,11 +25,27 @@
...
     """ Main entry point """
+    if current_app.test:
+        return
+    if config:
+        configure_app(current_app, yaml.safe_load(config))
+    else:
+        configure_app(current_app, {'DB_URI': db_uri,
+                                    'DB_NAME': db_name,
+                                    'DEBUG': debug,
```
Comment (hassanfa) [inline vogue/commands/base.py:46]:
```suggestion
                                    'DEBUG': flask_debug, 
```

File: vogue/commands/base.py (hunk 1 above)
Comment (mayabrandi) [reply vogue/commands/base.py]: 👍

**Comment by mayabrandi** [comment]
Thanks @hassanfa

**Comment by mayabrandi** [comment]
Performing test acording to desc above.
Running `vogue -c tmp.yaml load sample -d 1`
Skärmavbild 2020-04-15 kl  12 12 38

Running: `vogue load sample -d 1`

Skärmavbild 2020-04-15 kl  12 15 29

Can run the app locally, but having issues with the supervisor-ctrl on clinical-db
Skärmavbild 2020-04-15 kl  12 39 21
Skärmavbild 2020-04-15 kl  12 40 34

**Pull request merged**